### PR TITLE
[SPARK-5214][Test] Add a test to demonstrate EventLoop can be stopped in the event thread

### PR DESCRIPTION
**Pull request by zsxwing**

### CODE REVIEW

**Comment by srowen** [comment]
This was meant to go with https://github.com/apache/spark/pull/4016 ?

**Comment by zsxwing** [comment]
> This was meant to go with #4016 ?

I didn't consider this case when doing #4016. Although #4016 happens to work, I still think it's worth to add a test to prevent later changes from breaking it.

File: core/src/test/scala/org/apache/spark/util/EventLoopSuite.scala [hunk 1]
```diff
@@ -185,4 +185,22 @@ class EventLoopSuite extends FunSuite with Timeouts {
...
+        stop()
+      }
+
+      override def onError(e: Throwable): Unit = {
+      }
+
+    }
+    eventLoop.start()
+    eventLoop.post(1)
+    eventually(timeout(5 seconds), interval(200 millis)) {
```
Comment (rxin) [inline core/src/test/scala/org/apache/spark/util/EventLoopSuite.scala]: can we reduce the check interval to like 5ms? Ideally unit tests should run in a very short amount of time. Having to wait 200ms is pretty long. Would be great to change rest of this file also.

Comment (zsxwing) [inline core/src/test/scala/org/apache/spark/util/EventLoopSuite.scala]: Done

**Comment by rxin** [comment]
Thanks. Merging in master.

**Pull request closed without merging**